packages/volto/src/components/manage/Blocks/Search/hocs/withSearch.jsx

```diff
@@ -96,9 +96,26 @@ function normalizeState({
   const { types: facetWidgetTypes } =
     config.blocks.blocksConfig.search.extensions.facetWidgets;
 
+  // Here we are removing the QueryString of the Listing ones which is persent in the Facet
+  // because we already initialize the facet with those value.
+  const configuredFacets = facetSettings
+    ? facetSettings.map((facet) => facet?.field?.value)
+    : [];
+
+  let copyOfQuery = query.query ? [...query.query] : [];
+
+  for (let value of configuredFacets) {
```

Comment:
If I understand correctly, I think we can use a list filter for this whole code block, which would simplify it and make  easier to understand.